### PR TITLE
Conditionally rebuild `fs-xattr` before packaging DMG to fix dlopen issue

### PR DESCRIPTION
**Pull request by mokagio**
Builds on top of @fluiddot 's work on https://github.com/Automattic/studio/pull/299

I did some tests locally and found that if one runs `npm rebuild` in between the binary build and the DMG packaging, there is no `dlopen` issue in the `fs-xattr` call and the resulting DMG has the expected layout, as you can verify with the artifacts in [this build](https://buildkite.com/automattic/studio/builds/1661#01905da0-4d67-43b3-ae0f-6f59c828887f)

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -69,11 +69,6 @@ steps:
         agents:
           queue: mac
         command: |
-          echo "--- :macos: Install Rosetta to fix x64 xattr issue on arm64"
-          # Install Rosetta to try and address xattr architecture-based issue
-          # See https://buildkite.com/automattic/studio/builds/1543#01904ffb-8498-4d13-aa5a-e2b3c6dbb302
-          softwareupdate --install-rosetta --agree-to-license
-
```
Comment (mokagio) [inline .buildkite/pipeline.yml]: As the successful build for this commit shows, this step which I added as an initial experiment was unnecessary 

🧹 🧹 🧹 💨 💨 💨 

https://buildkite.com/automattic/studio/builds/1661#01905da0-4d67-43b3-ae0f-6f59c828887f

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -72,11 +72,28 @@ steps:
...
 
           .buildkite/commands/install-node-dependencies.sh
+
           node ./scripts/prepare-dev-build-version.mjs
 
           echo "--- :node: Building Binary"
           npm run make:macos-{{matrix}}
 
+          # Local trial and error show this needs to run before the DMG generation (obviously) but after the binary has been built
+          echo "--- :hammer: Rebuild fs-attr if necessary before genrationg DMG"
```
Comment (fluiddot) [inline .buildkite/pipeline.yml:82]:
```suggestion
          echo "--- :hammer: Rebuild fs-attr if necessary before generating DMG"
```

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -72,11 +72,28 @@ steps:
...
+          case {{matrix}} in
+            universal | x64)
+              echo "Rebuilding fs-xattr for {{matrix}} architecture"
+              npm rebuild fs-xattr --verbose --cpu universal
+              ;;
+            arm64)
+              echo "No need to rebuild fs-xattr because it works out of the box on Apple Silicon"
+              ;;
+            *)
+              echo "^^^ +++ Unexpected architcture {{matrix}}"
```
Comment (fluiddot) [inline .buildkite/pipeline.yml:92]:
```suggestion
              echo "^^^ +++ Unexpected architecture {{matrix}}"
```

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -72,11 +72,28 @@ steps:
...
 
           echo "--- :node: Building Binary"
           npm run make:macos-{{matrix}}
 
+          # Local trial and error show this needs to run before the DMG generation (obviously) but after the binary has been built
+          echo "--- :hammer: Rebuild fs-attr if necessary before genrationg DMG"
+          case {{matrix}} in
+            universal | x64)
+              echo "Rebuilding fs-xattr for {{matrix}} architecture"
+              npm rebuild fs-xattr --verbose --cpu universal
```
Comment (fluiddot) [inline .buildkite/pipeline.yml:86]: We probably don't need the logs from this command.

```suggestion
              npm rebuild fs-xattr --cpu universal
```

**Review by fluiddot — APPROVED** [review]
> I did some tests locally and found that if one runs `npm rebuild` in between the binary build and the DMG packaging, there is no `dlopen` issue in the `fs-xattr` call and the resulting DMG has the expected layout, as you can verify with the artifacts in [this build](https://buildkite.com/automattic/studio/builds/1661#01905da0-4d67-43b3-ae0f-6f59c828887f)

I experienced something similar when testing locally, but I didn't relate that the rebuild command needed to be executed between the build and packaging 😅. Good catch @mokagio !

**Comment by fluiddot** [comment]
I'll address the above comments in https://github.com/Automattic/studio/pull/299.

**Comment by wojtekn** [comment]
> I experienced something similar when testing locally, but I didn't relate that the rebuild command needed to be executed between the build and packaging 😅. Good catch @mokagio !

Nice! I also observed that sometimes it worked locally but couldn't figure out why.

**Comment by mokagio** [comment]
Thanks @fluiddot for adding the finishing polish. So many typos 🙃

**Pull request merged**